### PR TITLE
Feat: Docker Support for license DB

### LINKED ISSUE #63: Orchestration: LicenseDb needs docker support
-  [ ] Create Dockerfile for LicenseDb. For details regarding Go version and other dependencies, please refer to the go.mod file. Keep port configurable.
-  [ ] In the compose file, setup containers for application and database(Postgresql). 

For more setup details, please refer to README.md file.

### PR DESCRIPTION
**Pull request by Kaushl2208**
1. Dockerfile --> Multi Build stage for LicenseDB application
2. docker-compose.yml --> Orchestrate Postgres and Application (LicenseDB)

To Test:

```
docker compose build
docker compose up -d 
```

Closes #63

### CODE REVIEW

**Comment by Kaushl2208** [comment]
@GMishx Can you review :)

**Review by GMishx — APPROVED** [review]
Changes looks good, needs test.

Thanks for the contribution.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      - licensedb-network
+  postgres:
+    container_name: postgres
+    image: postgres:16
+    environment:
+      POSTGRES_PASSWORD: fossy
+      POSTGRES_USER: fossy
+      POSTGRES_DB: fossology
+    ports:
+      - "5432:5432"
```
Comment (GMishx) [inline docker-compose.yml:32]: It is not a good idea to expose the database port by default. Please comment the lines.

**Review by GMishx — CHANGES_REQUESTED** [review]

**Review by avinal — APPROVED** [review]
Tested, working as expected.

Possible next steps: GitHub Actions for pushing images to GitHub/Docker registry on push/release.

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Pull request merged**